### PR TITLE
Add cache header for image endpoint if link previews

### PR DESCRIPTION
**Pull request by juliusknorr**
Saves requests for the link preview images

### CODE REVIEW

File: lib/public/Collaboration/Reference/Reference.php [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?php
+
+namespace OCP\Collaboration\Reference;
+
+class Reference {
+
```
Comment (blizzz) [inline lib/public/Collaboration/Reference/Reference.php:6]: Where is this used? :eyes:

Comment (juliushaertl) [reply lib/public/Collaboration/Reference/Reference.php:6]: Whoops, sneaked in as we need this as a separate fix. Will remove from here and push a separate one with the reasoning.

**Review by blizzz — APPROVED** [review]

**Review by nickvergessen — APPROVED** [review]

**Review by CarlSchwan — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]

**Review by CarlSchwan — APPROVED** [review]

**Comment by juliushaertl** [comment]
Failure is unrelated

**Pull request merged**